dotcom-rendering/src/components/DiscussionApps.importable.tsx

```diff
@@ -1,23 +1,69 @@
-import { DiscussionResponseType } from '@guardian/bridget';
+import {
+	DiscussionResponseType,
+	GetUserProfileResponseType,
+} from '@guardian/bridget';
 import { useEffect, useState } from 'react';
 import { getDiscussionClient } from '../lib/bridgetApi';
-import type { Reader, UserProfile } from '../lib/discussion';
+import {
+	parseCommentResponse,
+	type Reader,
+	type UserProfile,
+} from '../lib/discussion';
 import type { CommentResponse } from '../lib/discussionApi';
 import { reportAbuse as reportAbuseWeb } from '../lib/discussionApi';
-import type { Result } from '../lib/result';
+import { error, type Result } from '../lib/result';
 import { Discussion, type Props as DiscussionProps } from './Discussion';
 
 type Props = Omit<DiscussionProps, 'user' | 'reportAbuseUnauthenticated'>;
 
-const onComment = async (): Promise<CommentResponse> => {
-	console.log('onComment');
-	return { kind: 'error', error: 'ApiError' };
-};
+const onComment = async (
+	discussionShortUrl: string,
+	body: string,
+): Promise<CommentResponse> =>
+	getDiscussionClient()
+		.comment(discussionShortUrl, body)
+		.then((apiResponse) => {
+			if (
+				apiResponse.__type ===
+				DiscussionResponseType.DiscussionResponseWithError
+			) {
+				return error('ApiError');
+			}
 
-const onReply = async (): Promise<CommentResponse> => {
-	console.log('onReply');
-	return { kind: 'error', error: 'ApiError' };
-};
+			if (apiResponse.response.status === 'error') {
+				return parseCommentResponse({
+					status: apiResponse.response.status,
+					errorCode: apiResponse.response.errorCode,
+				});
+			}
+
+			return parseCommentResponse({
+				status: apiResponse.response.status,
+				message: apiResponse.response.message,
+			});
+		});
+
+const onReply = async (
+	discussionShortUrl: string,
+	body: string,
+	parentCommentId: string,
+): Promise<CommentResponse> =>
+	getDiscussionClient()
+		.reply(discussionShortUrl, body, parentCommentId)
+		.then((apiResponse) => {
+			if (
+				apiResponse.__type ===
+				DiscussionResponseType.DiscussionResponseWithError
+			) {
+				return error('ApiError');
+			}
+
+			const parsedResponse = parseCommentResponse({
+				status: apiResponse.response.status,
+				message: apiResponse.response.message,
+			});
```

Comment:
There's no `"error"` status like there is for `comment`?

Reply:
I think I misunderstand your question, but yes - it's a few lines above:
https://github.com/guardian/dotcom-rendering/pull/11133/files#diff-5b1a8c963c0ac1496fdbf2131236a45ec6b9bbdcc862911a59e784196bad3864R33-R38